tools/db_bench_tool.cc

```diff
@@ -891,6 +891,8 @@ DEFINE_string(hdfs, "", "Name of hdfs environment. Mutually exclusive with"
               " --env_uri.");
 static rocksdb::Env* FLAGS_env = rocksdb::Env::Default();
 
+static std::shared_ptr<rocksdb::Env> env_guard;
```

Comment:
Is the destruction order the reversed? If it is the case, it is more natural to place env_guard before FLAGS_env .

Reply:
I believe so. However, even after env_guard is destructed, FLAGS_env pointing to invalid address should be fine because it won't be used anyway.
But I agree it sounds more natural, and I will update.